ESP8266Interface.cpp

```diff
@@ -20,21 +20,23 @@
 #include "mbed_debug.h"
 #include "nsapi_types.h"
 
-
-#ifndef MBED_CONF_ESP8266_TX
 #ifdef TARGET_FF_ARDUINO
+#ifndef MBED_CONF_ESP8266_TX
 #define MBED_CONF_ESP8266_TX D1
-#else
-#define MBED_CONF_ESP8266_TX NC
-#endif
 #endif
 
 #ifndef MBED_CONF_ESP8266_RX
-#ifdef TARGET_FF_ARDUINO
 #define MBED_CONF_ESP8266_RX D0
-#else
-#define MBED_CONF_ESP8266_RX NC
 #endif
+#endif /* TARGET_FF_ARDUINO */
+
+#if !defined MBED_CONF_ESP8266_TX || !defined MBED_CONF_ESP8266_RX
```

Comment:
Mbed OS by default compiles everything. Will you now break every board that does not specify these?

Reply:
True, unless the board conforms to Arduino form factor...

Reply:
Better to make the default constructor private in case TX&RX pins are NC.

Reply:
I agree.. if not Arduino, omit the default.